pkg/storage/stores/shipper/downloads/metrics.go

```diff
@@ -12,42 +10,19 @@ const (
 	statusSuccess = "success"
 )
 
-type downloadTableDurationMetric struct {
-	sync.RWMutex
-	gauge   prometheus.Gauge
-	periods map[string]float64
-}
-
-func (m *downloadTableDurationMetric) add(period string, downloadDuration float64) {
-	m.Lock()
-	defer m.Unlock()
-	m.periods[period] = downloadDuration
-
-	totalDuration := float64(0)
-	for _, dur := range m.periods {
-		totalDuration += dur
-	}
-
-	m.gauge.Set(totalDuration)
-}
-
 type metrics struct {
-	// metrics for measuring performance of downloading of files per period initially i.e for the first time
-	tablesDownloadDurationSeconds *downloadTableDurationMetric
-
+	queryTimeTableDownloadDurationSeconds  *prometheus.CounterVec
```

Comment:
I wonder if an histogram wouldn't be better here ?

Reply:
Not sure if that would add much value. Would we be interested in looking at a distribution? I think it would be rare and we can use `quantile_over_time` from the logline I have added [here](https://github.com/grafana/loki/pull/5484/files#diff-3b7a21722d5aa578d861ff5625cc9e841936e7a763e48d0009e196bfc691dd8eR298). wdyt?

Reply:
What's the PromQL you'll use with this ?

Reply:
something like `sum(quantile_over_time({container="index-gateway"} |= "downloaded index set at query time" | logfmt | unwrap duration(duration)[5m]) by (user_id)`. Makes sense?